src/nvim/drawscreen.c

```diff
@@ -1808,6 +1813,20 @@ static void win_update(win_T *wp, DecorProviders *providers)
       did_update = DID_NONE;
     }
 
+    // Redraw custom 'numberformat' in number column.
+    if (draw_num) {
+      const int nuw = wp->w_nrwidth_width;
+
+      win_redr_custom(wp, srow, false, false);
+      if (wp->w_nrwidth_width != nuw) {
+        // Number column got wider, start over.
```

Comment:
```suggestion
        // Number column width changed, start over.
```

Reply:
It can only get wider BTW, does that change your suggestion?

Reply:
Well the code has `if (wp->w_nrwidth_width != nuw) {`, but feel free to ignore the suggestion.

Reply:
Yeah true but the `win_redr_custom()` call only widens the number column if necessary. I guess `if (wp->w_nrwidth_width > nuw)` makes more sense then with the original comment.